tests/plugins/links/test_urlize.py

```diff
@@ -0,0 +1,231 @@
+
+    # Test trailing path specification.
+    (
+        'http://example.com/',
+        EXPECTED_PARAGRAPH_TEMPLATE % ('http://example.com/', 'http://example.com/')
+    ),
+    (
+        'http://example.com/my/path',
+        EXPECTED_PARAGRAPH_TEMPLATE % ('http://example.com/my/path', 'http://example.com/my/path')
+    ),
+    (
+        'http://example.com/my/path?param1=value1&param2=value2',
+        EXPECTED_PARAGRAPH_TEMPLATE % ('http://example.com/my/path?param1=value1&amp;param2=value2', 'http://example.com/my/path?param1=value1&amp;param2=value2')
+    ),
+
+    # Link positioned somewhere within the text, but around whitespace boundary.
+    (
+        'This is link myhost.example.com',
+        "<p>This is link " + EXPECTED_LINK_TEMPLATE % ('http://myhost.example.com', 'myhost.example.com') + "</p>"
+    ),
+    (
+        'myhost.example.com is the link',
+        "<p>" + EXPECTED_LINK_TEMPLATE % ('http://myhost.example.com', 'myhost.example.com') + " is the link</p>"
+    ),
+    (
+        'I have best myhost.example.com link ever',
+        "<p>I have best " + EXPECTED_LINK_TEMPLATE % ('http://myhost.example.com', 'myhost.example.com') + " link ever</p>"
+    ),
+    (
+        'I have best\nmyhost.example.com link ever',
+        "<p>I have best\n" + EXPECTED_LINK_TEMPLATE % ('http://myhost.example.com', 'myhost.example.com') + " link ever</p>"
+    ),
+]
+
+
+FIXTURE_NEGATIVE_MATCHES = [
+    # Incomplete FQDNs.
+    (
+        'example.',
+        '<p>example.</p>'
+    ),
+    (
+        '.example .com',
+        '<p>.example .com</p>'
+    ),
+
+    # localhost as part of another word.
+    (
+        'localhosts',
+        '<p>localhosts</p>'
+    ),
+
+    # Invalid FQDNs.
+    (
+        'example-.com',
+        '<p>example-.com</p>'
+    ),
+    (
+        '-example.com',
+        '<p>-example.com</p>'
+    ),
+    (
+        'my.-example.com',
+        '<p>my.-example.com</p>'
+    ),
+
+    # Invalid IPv6 patterns.
+    (
+        '1:2:3:4:5:6:7:8:a',  # Use :a, because using a number would match as optional port
+        '<p>1:2:3:4:5:6:7:8:a</p>',
+    ),
+    (
+        '1::2::3',
+        '<p>1::2::3</p>',
+    ),
+    (
+        '::::1',
+        '<p>::::1</p>',
+    ),
+    (
+        '1::::',
+        '<p>1::::</p>',
+    ),
+
+    # Invalid IPv4 patterns.
+    (
+        '1.2.3.4.5',
+        '<p>1.2.3.4.5</p>',
+    ),
+
+]
+
+
+class TestUrlizeExtension:
+
+    def setup_method(self):
+        self.md = markdown.Markdown(extensions=[UrlizeExtension()])
+
+    @pytest.mark.parametrize("markdown_text, expected_output", FIXTURE_POSITIVE_MATCHES)
+    def test_positive_matches(self, markdown_text, expected_output):
+        assert self.md.convert(markdown_text) == expected_output
+
+    @pytest.mark.parametrize("markdown_text, expected_output", FIXTURE_NEGATIVE_MATCHES)
+    def test_negative_matches(self, markdown_text, expected_output):
+        assert self.md.convert(markdown_text) == expected_output
```

Comment:
See comment above, use `self.assertEqual` instead? This is what all the other tests use.

Reply:
Actually, when you are running `pytest`, it will catch these assertion errors and display the problems just as cleanly as `self.assertEqual`. Maybe even better. I don't want to say that it's necessarily better than the other, because it implies fixing up the remaining codebase or having some policy.. I would think both are fine :)

Reply:
If nothing else in the test code uses this, I'll convert this into `unittest` test. Consistency is good. Or, as I like to put it - you want to introduce something new? Then go ahead and do it _everywhere_ :)

Reply:
Went for ease of use instead of consistency. I'll trump over my own words and not do it everywhere, though...